src/quotamodel.c

```diff
@@ -1161,25 +1136,30 @@ delete_from_table_size_map(char *str)
 	                 "delete from diskquota.table_size "
 	                 "where (tableid, segid) in ( SELECT * FROM deleted_table );",
 	                 str);
+	SPI_connect_my(&state);
```

Comment:
Aren't these functions called multiple times from inside flush_to_table_size()? Wouldn't this commit the transaction prematurely?

Reply:
> Aren't these functions called multiple times from inside flush_to_table_size()? Wouldn't this commit the transaction prematurely?

No, because the commit in the `SPI_finish_my` function is only made if the `SPI_connect_my` function was called outside of the transaction. And the `flush_to_table_size` function is called in a transaction that was already explicitly started in the `refresh_disk_quota_usage` function.